pkg/kubelet/pleg/evented.go

```diff
@@ -0,0 +1,471 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pleg
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	internalapi "k8s.io/cri-api/pkg/apis"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	"k8s.io/klog/v2"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/utils/clock"
+)
+
+// IsEventedPLEGInUse indicates whether evented PLEG is in use. Even after enabling
+// the Evented PLEG feature gate, there could be several reasons it may not be in use.
+// e.g. Streaming data issues from the runtime or the runtime does not implement the
+// container events stream.
+var IsEventedPLEGInUse = false
```

Comment:
Maybe other goroutines should not access this variable directly, as if pleg goroutine change the value, there may be data race error while  other goroutines like podworker read this

Reply:
That variable was added to indicate pod worker if the evented pleg is in use or not. We are going to add some more experimental code so I am moving this PR in `WIP` state.